### ISSUE TITLE
mini xgboost issue, problem with softprob

### ISSUE THREAD

**Issue by berndbischl** [mlr-org/mlr#706]
this was reported

```

Thank you for your mlr package !
I’m using it for an xgboost classifier with multi:softprob objective.
I get an error with only two classes : Error in y[, 1L] = 1 - p :  le nombre d'objets à remplacer n'est pas multiple de la taille du remplacement

In a new classifier classif.xgboost2, I modified the predictLearner.classif.xgboost2 function and replace 

  if (nc == 2L) {

      y = matrix(0, ncol = 2, nrow = nrow(.newdata))
      colnames(y) = td$class.levels
      y[, 1L] = 1-p
      y[, 2L] = p
    }

by 

  if (nc == 2L) {
    if(.learner$par.vals$objective == "multi:softprob") {
       nc = length(td$class.levels)
       y = matrix(p,nc,length(p)/nc)
       y = t(y)
       colnames(y) = td$class.levels
    } else 
      y = matrix(0, ncol = 2, nrow = nrow(.newdata))
      colnames(y) = td$class.levels
      y[, 1L] = 1-p
      y[, 2L] = p
    }

and it works fine,

```

**Comment by berndbischl** [comment]
I already have a partial fix in branch
fix_xgboost

**Comment by berndbischl** [comment]
also note my comments there, we should probably report this to xgboost

**Comment by berndbischl** [comment]
also we should reread the paramset, not all params are tuneable for example (printers, nthread etc)

**Comment by pfistfl** [comment]
Additionally some params that are not documented in the r-package are missing, ```max_delta_step```
for example. 
Those are documented in https://xgboost.readthedocs.org/en/latest/parameter.html though

**Comment by berndbischl** [comment]
@Coorsaa 
Please take this over

**Comment by ja-thomas** [comment]
```
train(makeLearner("classif.xgboost"), iris.task)
train(makeLearner("classif.xgboost", objective = "multi:softprob"), iris.task)
train(makeLearner("classif.xgboost", objective = "multi:softmax"), iris.task)
train(makeLearner("classif.xgboost", objective = "binary:logistic"), iris.task) #error
train(makeLearner("classif.xgboost"), binaryclass.task)
train(makeLearner("classif.xgboost", objective = "multi:softprob"), binaryclass.task)
train(makeLearner("classif.xgboost", objective = "multi:softmax"), binaryclass.task)

resample(makeLearner("classif.xgboost"), iris.task, resampling = hout)
resample(makeLearner("classif.xgboost", objective = "multi:softprob"), iris.task, resampling = hout)
resample(makeLearner("classif.xgboost", objective = "multi:softmax"), iris.task, resampling = hout)
resample(makeLearner("classif.xgboost", objective = "binary:logistic"), iris.task, resampling = hout) #error
resample(makeLearner("classif.xgboost"), binaryclass.task, resampling = hout)
resample(makeLearner("classif.xgboost", objective = "multi:softprob"), binaryclass.task, resampling = hout)
resample(makeLearner("classif.xgboost", objective = "multi:softmax"), binaryclass.task, resampling = hout)
```

**Comment by Coorsaa** [comment]
while training and predicting is already working (branch fix_706) I still get errors for objective = multi:softmax for resampling and benchmarking, see below:

````
lrn = makeLearner("classif.xgboost")
set.seed(1)

mod1 = train(makeLearner("classif.xgboost"), iris.task)
mod2 = train(makeLearner("classif.xgboost", objective = "multi:softprob"), iris.task)
mod3 = train(makeLearner("classif.xgboost", objective = "multi:softmax"), iris.task)
mod4 = train(makeLearner("classif.xgboost", objective = "binary:logistic"), iris.task) #expect error
mod5 = train(makeLearner("classif.xgboost"), binaryclass.task)
mod6 = train(makeLearner("classif.xgboost", objective = "multi:softprob"), binaryclass.task)
mod7 = train(makeLearner("classif.xgboost", objective = "multi:softmax"), binaryclass.task)

pred1 = predict(mod1, newdata = getTaskData(iris.task))
pred2 = predict(mod2, newdata = getTaskData(iris.task))
pred3 = predict(mod3, newdata = getTaskData(iris.task))
pred5 = predict(mod5, newdata = getTaskData(binaryclass.task))
pred6 = predict(mod6, newdata = getTaskData(binaryclass.task))
pred7 = predict(mod7, newdata = getTaskData(binaryclass.task))


res1 = resample(makeLearner("classif.xgboost"), iris.task, resampling = hout)
res2 = resample(makeLearner("classif.xgboost", objective = "multi:softprob"), iris.task, resampling = hout)
res3 = resample(makeLearner("classif.xgboost", objective = "multi:softmax"), iris.task, resampling = hout)
res5 = resample(makeLearner("classif.xgboost"), binaryclass.task, resampling = hout)
res6 = resample(makeLearner("classif.xgboost", objective = "multi:softprob"), binaryclass.task, resampling = hout)
res7 = resample(makeLearner("classif.xgboost", objective = "multi:softmax"), binaryclass.task, resampling = hout)

ben1 = benchmark(makeLearner("classif.xgboost"), iris.task)
ben2 = benchmark(makeLearner("classif.xgboost", objective = "multi:softprob"), iris.task)
ben3 = benchmark(makeLearner("classif.xgboost", objective = "multi:softmax"), iris.task)
ben5 = benchmark(makeLearner("classif.xgboost"), binaryclass.task)
ben6 = benchmark(makeLearner("classif.xgboost", objective = "multi:softprob"), binaryclass.task)
ben7 = benchmark(makeLearner("classif.xgboost", objective = "multi:softmax"), binaryclass.task)
````
